pkg/test/fakes/imageregistry/main.go

```diff
@@ -0,0 +1,113 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"crypto/tls"
+	"encoding/base64"
+	"flag"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"istio.io/pkg/log"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	containerregistry "github.com/google/go-containerregistry/pkg/registry"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+)
+
+var port = flag.Int("port", 1338, "port to run registry on")
+var images = flag.String("images", "", "comma separated list of images that will be preloaded and served")
```

Comment:
should we bake them into the docker image? so startup is improved?

Reply:
I am planning to test with images generated from istio/proxy (e.g. attributes gen), which could often change, so that's why I am making it configured dynamically here.

Reply:
If they are already pushed to some registry I don't really get what we need this for then...? Just to add user and passwd check? If just that it seems like we could just 301 redirect to the real registry (not 100% sure its possible but I think it is)?

Reply:
Updated to redirect.